lib/grandpa/message_handler.go

```diff
@@ -37,6 +40,8 @@ func NewMessageHandler(grandpa *Service, blockState BlockState, telemetryMailer
 	}
 }
 
+//nolint
+// TODO: NotificationMessage is used at places. But NotificationMessage we return is always nil.
```

Comment:
the message that's returned is the response that's sent to the peer, `handleCatchUpRequest` should return a response potentially

Reply:
change the `handleMessage` return params